app/src/main/java/org/lineageos/selfie/ui/CountDownView.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2014 The Android Open Source Project
+ * Copyright (C) 2022 The LineageOS Project
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.lineageos.selfie.ui
+
+import android.content.Context
+import android.graphics.Rect
+import android.os.Handler
+import android.os.Looper
+import android.os.Message
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import android.widget.TextView
+import androidx.annotation.IntRange
+import org.lineageos.selfie.R
+
+/**
+ * This class manages the looks of the countdown.
+ */
+class CountDownView(context: Context, attrs: AttributeSet?) : FrameLayout(
+    context, attrs
+) {
+    private val mRemainingSecondsView by lazy {
+        findViewById<TextView>(R.id.remaining_seconds)
+    }
+    private var mRemainingSecs = 0
+    private lateinit var mListener: () -> Unit
+    private val mHandler = MainHandler(Looper.getMainLooper())
+    private val mPreviewArea = Rect()
+
+    /**
+     * Returns whether countdown is on-going.
+     */
+    private val isCountingDown = mRemainingSecs > 0
```

Comment:
I think this should be inside a get() function, unless Kotlin treats that assignment as a unit but I doubt that

Reply:
We aren't using this bit, but yeah, fixed.